app/controllers/developer_controller.rb

```diff
@@ -13,7 +13,10 @@ def update_sites
 
   def production_log
     @log = if params[:grep].present?
-             `tail -n 10000 log/production.log | grep -E '#{params[:grep]}' -C #{params[:context]} --color=never`
+             unless params[:grep].instance_of?(String) && params[:context].instance_of?(Integer)
```

Comment:
Have you tested this? I generally prefer to avoid `instance_of?` checks and instead try to attempt a coercion (`to_s` or `to_i`. Also, this seems like it would break grep search since I don't see you converting `context` to an integer anywhere and I believe params are all strings by default.